tools/ci/check-changelog.js

```diff
@@ -0,0 +1,46 @@
+const fs = require("fs");
+const proc = require("child_process");
+
+const skipChangelogInfix = "[no-changelog]";
+const changelogPath = process.argv[2];
+const baseRef = process.argv[3];
+
+function wasChangelogModified() {
+  const diffArgs = [
+    "--no-pager",
+    "diff",
+    "--exit-code",
+    baseRef,
+    "--",
+    changelogPath,
+  ];
+  const result = proc.spawnSync("git", diffArgs);
+  const exitCode = result.status;
+  console.log(result.stdout.toString("utf-8"));
+  const noDifference = exitCode == 0;
+  return !noDifference;
+}
+
+function isChangelogSkipped() {
+  const logArgs = ["--no-pager", "log", "HEAD~3...HEAD", "--pretty=oneline"];
+  const output = proc.spawnSync("git", logArgs).stdout.toString("utf-8");
+  const containsSkipCommit = output.indexOf(skipChangelogInfix) >= 0;
+  return containsSkipCommit;
+}
+
+if (wasChangelogModified()) {
+  console.log("Changelog was changed");
+  process.exit(0);
+} else {
+  console.log("No changes to the changelog");
+  if (isChangelogSkipped()) {
+    console.log(
+      "But one of the commits within the PR includes " +
+        skipChangelogInfix +
+        ", so the check is skipped."
+    );
+    process.exit(0);
+  } else {
+    process.exit(1);
+  }
+}
```

Comment:
Have you made sure that this being missing fails PR CI?

Reply:
[Yes](https://github.com/enso-org/enso-staging/runs/2401632708).

It was done some time ago, but the relevant code did not change since.